thewalrus/reference.py

```diff
@@ -375,3 +376,51 @@ def rspm(s):
     """
     gen = rpmp(s)
     return chain(*(splitter(i) for i in gen))
+
+
+def mtl(A):
+    """Returns the Montrealer of an NxN matrix and an N-length vector.
+
+    Args:
+        A (array): an NxN array of even dimensions. Can be symbolic.
+
+    Returns:
+        np.float64, np.complex128 or sympy.core.add.Add: the Montrealer of matrix A.
+    """
+    n,_ = A.shape
+    net_sum = 0
+
+    for s in rpmp(range(n)):
+        net_prod = 1
+        for a in s:
+            a = sorted(a)
+            net_prod *= A[a[0],a[1]]
+            
+        net_sum += net_prod
+
+    return net_sum
+
+
+def lmtl(A, zeta):
```

Comment:
Also, we should follow the convention of the `hafnian` function above, where instead of having `hafnian` and `loop_hafnian` we have a single function with the optional argument `loops`: 
![image](https://github.com/XanaduAI/thewalrus/assets/991946/1d9a696e-2f3b-448f-a929-7f257677bfe8)

Reply:
Fixed.